frame/composable-maths/src/dex/price.rs

```diff
@@ -0,0 +1,66 @@
+use composable_traits::{
+	defi::Rate,
+	math::{SafeArithmetic, SafeMul},
+};
+use frame_support::ensure;
+use sp_runtime::{
+	traits::{AtLeast32Bit, Convert as ConvertTrait, Saturating, Zero},
+	ArithmeticError, DispatchError, FixedPointNumber, SaturatedConversion,
+};
+
+/// Uniswap V2 TWAP
+/// https://docs.uniswap.org/protocol/V2/concepts/core-concepts/oracles
+pub fn compute_initial_price_cumulative<Convert, Balance>(
+	current_exchange_rate: Rate,
+) -> Result<Balance, DispatchError>
+where
+	Convert: ConvertTrait<u128, Balance>,
+	Balance: Zero + SafeArithmetic,
+{
+	compute_next_price_cumulative::<Convert, Balance, _>(
+		0_u64,
+		Balance::zero(),
+		1_u64,
+		current_exchange_rate,
+	)
+	.map(|(_, x)| x)
+}
+
+pub fn compute_next_price_cumulative<Convert, Balance, Timestamp>(
+	previous_timestamp: Timestamp,
+	previous_price_cumulative: Balance,
+	current_timestamp: Timestamp,
+	current_exchange_rate: Rate,
+) -> Result<(Timestamp, Balance), DispatchError>
+where
+	Balance: SafeArithmetic,
+	Timestamp: AtLeast32Bit + SaturatedConversion + Saturating + Ord + Copy,
+	Convert: ConvertTrait<u128, Balance>,
+{
+	ensure!(current_timestamp > previous_timestamp, ArithmeticError::Underflow);
```

Comment:
`ArithmeticError::Underflow` Seems the wrong error to me, perhaps use the Other vaiant with a more descriptive message?

Reply:
@KaiserKarel any suggestions? I don't think any thing from DispatchError fits. Perhaps add `StaleRequrest` to pablo's Error?